action/ensure.go

```diff
@@ -33,6 +35,25 @@ func EnsureConfig() *cfg.Config {
 		msg.Die("Failed to parse %s: %s", yamlpath, err)
 	}
 
+	b := filepath.Dir(yamlpath)
+	buildContext, err := util.GetBuildContext()
+	if err != nil {
+		msg.Die("Failed to build an import context while ensuring config: %s", err)
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		msg.Err("Unable to get the current working directory")
+	} else {
+		// Determining a package name requires a relative path
+		b, err = filepath.Rel(b, cwd)
+		if err == nil {
+			name := buildContext.PackageName(b)
+			if name != conf.Name {
+				msg.Warn("The name listed in the config file (%s) does not match the current location (%s)", conf.Name, name)
+			}
```

Comment:
@technosophos An error should never happen. The reason I didn't do anything was I wanted to keep the user experience sane. I think I'll put a debugging message in.